File: src/main/rest/repository/GoalRepository.js
import Goal from '../model/Goal'

const GoalRepository = {


    insertAll : (txn, goals) => {
        let query = 'insert into goal(id, purpose_id, name, description, start_date, end_date, color, cycle, briefing_count, last_briefing_date) values';
        goals.forEach((goal, index) => {
            query += `(${goal.id} , ${goal.purposeId}, '${goal.name}', '${goal.description}', '${goal.startDate.toString()}', '${goal.endDate.toString()}', '${goal.color}', '${goal.cycle}', ${goal.briefingCount}, ${goal.lastBriefingDate ? '\''+goal.lastBriefingDate.toString()+'\'' : null} )`
            query += index + 1 != goals.length ? ', ' : '';
        });

   
        return new Promise((resolve, reject) => {
            txn.executeSql(
                query,
                [],
                resolve,
                reject
            );
        })

    },

    insert : (txn, goal) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'insert into goal(id, purpose_id, name, description, start_date, end_date, color, cycle, briefing_count, last_briefing_date) values(?,?,?,?,?,?,?,?,?,?)',
                [goal.id, goal.purposeId, goal.name, goal.description, goal.startDate.toString(), goal.endDate.toString(), goal.color, goal.cycle, goal.briefingCount, goal.lastBriefingDate ? goal.lastBriefingDate.toString() : null],
                resolve,
                reject
            );
        })
    },
    
    selectByHeaderIdAndStat : (txn, purposeId, stat) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'select id, purpose_id, name, description, start_date, end_date, color, cycle, briefing_count, last_briefing_date from goal where purpose_id = ?',
                [purposeId, stat],
                (res) => {
                    let data = [];
                    for(let i =0; i < res.rows.length; i++){
                        const row = res.rows.item(i);
                        data.push(new Goal(row.id, row.purpose_id, row.name, row.description, row.start_date, row.end_date, row.color, row.cycle, row.briefing_count, row.last_briefing_date))
                    }
                    resolve(data);
                },
                reject)
        })
    },

    selectByPurposeId : (txn, purposeId) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'select id, purpose_id, name, description, start_date, end_date, color, cycle, briefing_count, last_briefing_date from goal where purpose_id = ?',
                [purposeId],
                (res) => {
                    let data = [];
                    for(let i =0; i < res.rows.length; i++){
                        const row = res.rows.item(i);
                        data.push(new Goal(row.id, row.purpose_id, row.name, row.description, row.start_date, row.end_date, row.color, row.cycle, row.briefing_count, row.last_briefing_date))
                    }
                    resolve(data);
                },
                reject
            );
        })
    },

    selectInPurposeId : (txn, purposeIdList) => {
        let ext = purposeIdList[0];
        for(let i = 1; i< purposeIdList.length; i++){
            ext += "," + purposeIdList[i];
        }
        return new Promise((resolve, reject) => {
            txn.executeSql(
                `select id, purpose_id, name, description, start_date, end_date, color, cycle, briefing_count, last_briefing_date from goal where purpose_id in (${ext})`,
                [],
                (res) => {
                    let data = [];
                    for(let i =0; i < res.rows.length; i++){
                        const row = res.rows.item(i);
                        data.push(new Goal(row.id, row.purpose_id, row.name, row.description, row.start_date, row.end_date, row.color, row.cycle, row.briefing_count, row.last_briefing_date))
                    }
                    resolve(data);
                },
                reject
            );
        })   
    },

    deleteByPurposeId : (txn, purposeId) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'delete from goal where purpose_id = ?',
                [purposeId],
                resolve,
                reject
            )
        })
    },

    updateByKey : (txn, purposeId, id, goal) => {
        console.log(goal);
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'update goal set stat = ?, briefing_count = ?, last_briefing_date = ? where purpose_id = ? and id = ?',
                [goal.stat, goal.briefingCount, goal.lastBriefingDate ? goal.lastBriefingDate.toString() : null, purposeId, id],
                resolve,
                reject
            );
        })
    }

}

export default GoalRepository;

File: android/app/src/main/java/com/downfall/caterplanner/rest/repository/GoalRepository.java
package com.downfall.caterplanner.rest.repository;

import android.database.Cursor;

import com.downfall.caterplanner.rest.model.Goal;
import com.downfall.caterplanner.rest.model.State;
import com.downfall.caterplanner.util.DateUtil;

import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

public class GoalRepository extends BaseRepository {

    public void insert(Goal goal){
        final String sql =
                "insert into goal values(?,?,?,?,?,?,?)";
        db.execSQL(sql, new String[]{
                String.valueOf(goal.getPurposeId()),
                String.valueOf(goal.getId()),
                String.valueOf(goal.getName()),
                String.valueOf(goal.getStartDate()),
                String.valueOf(goal.getEndDate()),
                String.valueOf(goal.getColor()),
                String.valueOf(goal.getStat())
        });
    }

    public List<Goal> selectByHeaderIdAndStat(long purposeId, State stat) throws ParseException {
        final String sql =
                "select purpose_id, id, name, start_date, end_date, color, stat " +
                        "from goal " +
                        "where purpose_id = ? and stat = ?";
        Cursor c = db.rawQuery(sql, new String[]{String.valueOf(purposeId), String.valueOf(stat)});
        List<Goal> goals = new ArrayList<>();
        while(c.moveToNext()){
            goals.add(
                    Goal.builder()
                            .purposeId(c.getLong(0))
                            .id(c.getInt(1))
                            .name(c.getString(3))
                            .startDate(DateUtil.parseToDate(c.getString(4)))
                            .endDate(DateUtil.parseToDate(c.getString(5)))
                            .color(c.getString(6))
                            .stat(State.findByValue(c.getInt(7)))
                            .build()
            );
        }
        return goals;
    }

    public List<Goal> selectByPurposeId(long purposeId) throws ParseException {
        final String sql =
                "select purpose_id, id, name, start_date, end_date, color, stat " +
                        "from goal " +
                        "where purpose_id = ?";
        Cursor c = db.rawQuery(sql, new String[]{String.valueOf(purposeId)});
        List<Goal> goals = new ArrayList<>(c.getCount());
        while(c.moveToNext()){
           goals.add(
                   Goal.builder()
                           .purposeId(c.getLong(0))
                           .id(c.getInt(1))
                           .name(c.getString(3))
                           .startDate(DateUtil.parseToDate(c.getString(4)))
                           .endDate(DateUtil.parseToDate(c.getString(5)))
                           .color(c.getString(6))
                           .stat(State.findByValue(c.getInt(7)))
                           .build()
           );
        }
        return goals;
    }

    public void deleteByPurposeId(long purposeId){
        final String sql =
                "delete from goal where purpose_id = ?";
        db.execSQL(sql, new String[]{String.valueOf(purposeId)});
    }

    public void updateStatByKey(long purposeId, int id, State state){
        final String sql =
                "update goal " +
                        "set stat = ? " +
                        "where purpose_id = ? and id = ?";
        db.execSQL(sql, new String[]{
                String.valueOf(state.getValue()),
                String.valueOf(purposeId),
                String.valueOf(id)
        });
    }




}
